src/docfx/lib/YamlUtility.cs

```diff
@@ -103,18 +105,19 @@ public static (List<Error>, JToken) Deserialize(TextReader reader, bool nullVali
 
             if (nullValidation)
             {
-                var mappings = new JTokenSourceMap();
-                var (nullErrors, token) = ToJson(stream.Documents[0].RootNode, mappings).ValidateNullValue(mappings);
+                var root = stream.Documents[0].RootNode;
+                var (nullErrors, token) = PopulateLineInfoToJToken(ToJson(root), root).ValidateNullValue();
                 errors.AddRange(nullErrors);
                 return (errors, token);
             }
             else
             {
-                return (errors, ToJson(stream.Documents[0].RootNode));
+                var token = ToJson(stream.Documents[0].RootNode);
```

Comment:
populate line info even without null validation?

Reply:
Because we are gonna need the line info for null value error, so we should prepare line info before that

---
In reply to: [203930545](https://github.com/dotnet/docfx/pull/3100#discussion_r203930545) [](ancestors = 203930545)